sql/plan/showvariables.go

```diff
@@ -80,8 +85,15 @@ func (*ShowVariables) Children() []sql.Node { return nil }
 // The function returns an iterator for filtered variables (based on like pattern)
 func (sv *ShowVariables) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
 	var rows []sql.Row
+	var sysVars map[string]interface{}
+
+	if sv.global {
+		sysVars = ctx.GetAllGlobalVariables()
+	} else {
+		sysVars = ctx.GetAllSessionVariables()
+	}
```

Comment:
Instead of putting the new `GetAllGlobalVariables` function on the context/session object... it would probably make more sense as a new function directly in `SystemVariables`. It makes sense to ask the session for the all the current session variables, but for global variables, I think it makes a little more sense to go straight to `SystemVariables` and ask for all global variables.